pxf/pxf-hive/src/main/java/org/apache/hawq/pxf/plugins/hive/utilities/HiveUtilities.java

```diff
@@ -102,80 +103,49 @@ public static Table getHiveTable(HiveMetaStoreClient client, Metadata.Item itemN
      * <li>{@code decimal(precision, scale) -> numeric(precision, scale)}</li>
      * <li>{@code varchar(size) -> varchar(size)}</li>
      * <li>{@code char(size) -> bpchar(size)}</li>
+     * <li>{@code array<dataType> -> text}</li>
+     * <li>{@code map<keyDataType, valueDataType> -> text}</li>
+     * <li>{@code struct<field1:dataType,...,fieldN:dataType> -> text}</li>
+     * <li>{@code uniontype<...> -> text}</li>
      * </ul>
      *
-     * @param hiveColumn hive column schema
+     * @param hiveColumn
+     *            hive column schema
      * @return field with mapped HAWQ type and modifiers
-     * @throws UnsupportedTypeException if the column type is not supported
+     * @throws UnsupportedTypeException
+     *             if the column type is not supported
+     * @see EnumHiveToHawqType
      */
     public static Metadata.Field mapHiveType(FieldSchema hiveColumn) throws UnsupportedTypeException {
         String fieldName = hiveColumn.getName();
-        String hiveType = hiveColumn.getType();
-        String mappedType;
-        String[] modifiers = null;
+        String hiveType = hiveColumn.getType(); // Type name and modifiers if any
+        String hiveTypeName; // Type name
+        String[] modifiers = null; // Modifiers
+        EnumHiveToHawqType hiveToHawqType = EnumHiveToHawqType.getHiveToHawqType(hiveType);
+        EnumHawqType hawqType = hiveToHawqType.getHawqType();
 
-        // check parameterized types:
-        if (hiveType.startsWith("varchar(") ||
-                hiveType.startsWith("char(")) {
-            String[] toks = hiveType.split("[(,)]");
-            if (toks.length != 2) {
-                throw new UnsupportedTypeException( "HAWQ does not support type " + hiveType + " (Field " + fieldName + "), " +
-                        "expected type of the form <type name>(<parameter>)");
-            }
-            mappedType = toks[0];
-            if (mappedType.equals("char")) {
-                mappedType = "bpchar";
-            }
-            modifiers = new String[] {toks[1]};
-        } else if (hiveType.startsWith("decimal(")) {
-            String[] toks = hiveType.split("[(,)]");
-            if (toks.length != 3) {
-                throw new UnsupportedTypeException( "HAWQ does not support type " + hiveType + " (Field " + fieldName + "), " +
-                        "expected type of the form <type name>(<parameter>,<parameter>)");
+        if (hiveToHawqType.getSplitExpression() != null) {
+            String[] tokens = hiveType.split(hiveToHawqType.getSplitExpression());
+            hiveTypeName = tokens[0];
```

Comment:
Since you already execute similar logic to parse the tokens in `getHiveToHawqType()` why not set `tokens` as a field on `EnumHiveToHawqType` to consolidate the code to one place?